aviary/subsystems/propulsion/propeller/propeller_performance.py

```diff
@@ -305,6 +339,130 @@ def compute_partials(self, inputs, J):
                 np.sqrt(helical_mach * helical_mach - mach * mach)
 
 
+class AreaSquareRatio(om.ExplicitComponent):
+    """
+    Compute the area ratio nacelle and propeller with a maximum 0.5.
+    """
+
+    def initialize(self):
+        self.options.declare("num_nodes", types=int)
+        self.options.declare('smooth_sqa', default=True, types=bool)
+        self.options.declare('alpha', default=100.0, types=float)
+
+    def setup(self):
+        nn = self.options["num_nodes"]
+        arange = np.arange(self.options["num_nodes"])
+        self.add_input("DiamNac", val=0.0, units='ft')
+        self.add_input("DiamProp", val=0.0, units='ft')
+
+        self.add_output('sqa_array', val=np.zeros(nn), units='unitless')
+
+        self.declare_partials("sqa_array",
+                              [
+                                  "DiamNac",
+                                  "DiamProp",
+                              ],
+                              rows=arange, cols=np.zeros(nn))
+
+    def compute(self, inputs, outputs):
+        nn = self.options["num_nodes"]
+        diamNac = inputs["DiamNac"]
+        diamProp = inputs["DiamProp"]
+        sqa = diamNac**2 / diamProp**2
+
+        smooth = self.options["smooth_sqa"]
+        if smooth:
+            alpha = self.options['alpha']
+            sqa = smooth_min(sqa, 0.50, alpha)
+        else:
+            sqa = np.minimum(sqa, 0.50)
+        outputs["sqa_array"] = np.ones(nn) * sqa
+
+    def compute_partials(self, inputs, partials):
+        nn = self.options["num_nodes"]
+        diamNac = inputs["DiamNac"]
+        diamProp = inputs["DiamProp"]
+        sqa = diamNac**2 / diamProp**2
+
+        dSQA_dNacDiam = 2 * diamNac / diamProp**2
+        dSQA_dPropDiam = -2 * diamNac**2 / diamProp**3
+
+        smooth = self.options["smooth_sqa"]
+        if smooth:
+            alpha = self.options['alpha']
+            sqa = smooth_min(sqa, 0.50, alpha)
+            dSQA_dNacDiam = d_smooth_min(sqa, alpha) * dSQA_dNacDiam
+            dSQA_dPropDiam = d_smooth_min(sqa, alpha) * dSQA_dPropDiam
```

Comment:
here (and potentially elsewhere) `d_smooth_min(sqa, alpha)` means that when `def d_smooth_min(x, b, alpha=100.0)` is called, sqa will be used as x, alpha will be used as b. I think this needs to be `d_smooth_min(sqa, .5, alpha)`